generator/src/main/java/org/apache/kafka/message/MessageDataGenerator.java

```diff
@@ -1581,56 +1570,56 @@ private void generateVariableLengthFieldSize(FieldSpec field,
                     } else {
-                        buffer.printf("_size += _arraySize;%n");
+                        buffer.printf("_size.add(_arraySize);%n");
                     }
                 } else if (field.type().isBytes()) {
+                    buffer.printf("MessageSize _bytesSize = new MessageSize();%n");
                     if (field.zeroCopy()) {
-                        buffer.printf("int _bytesSize = %s.remaining();%n", field.camelCaseName());
+                        buffer.printf("_bytesSize.addZeroCopyBytes(%s.remaining());%n", field.camelCaseName());
                     } else {
-                        buffer.printf("int _bytesSize = %s.length;%n", field.camelCaseName());
+                        buffer.printf("_bytesSize.addBytes(%s.length);%n", field.camelCaseName());
                     }
                     VersionConditional.forVersions(fieldFlexibleVersions(field), possibleVersions).
                         ifMember(__ -> {
                             headerGenerator.addImport(MessageGenerator.BYTE_UTILS_CLASS);
                             if (field.zeroCopy()) {
-                                buffer.printf("_bytesSize += " +
-                                        "ByteUtils.sizeOfUnsignedVarint(%s.remaining() + 1);%n", field.camelCaseName());
+                                buffer.printf("_bytesSize.addBytes(" +
+                                        "ByteUtils.sizeOfUnsignedVarint(%s.remaining() + 1));%n", field.camelCaseName());
                             } else {
-                                buffer.printf("_bytesSize += ByteUtils.sizeOfUnsignedVarint(%s.length + 1);%n",
+                                buffer.printf("_bytesSize.addBytes(ByteUtils.sizeOfUnsignedVarint(%s.length + 1));%n",
                                     field.camelCaseName());
                             }
                         }).
                         ifNotMember(__ -> {
-                            buffer.printf("_bytesSize += 4;%n");
+                            buffer.printf("_bytesSize.addBytes(4);%n");
                         }).
                         generate(buffer);
                     if (tagged) {
                         headerGenerator.addImport(MessageGenerator.BYTE_UTILS_CLASS);
-                        buffer.printf("_size += _bytesSize + ByteUtils.sizeOfUnsignedVarint(_bytesSize);%n");
-                    } else {
-                        buffer.printf("_size += _bytesSize;%n");
+                        buffer.printf("_size.addBytes(ByteUtils.sizeOfUnsignedVarint(_bytesSize.totalSize()));%n");
                     }
+                    buffer.printf("_size.add(_bytesSize);%n");
                 } else if (field.type().isRecords()) {
-                    buffer.printf("int _recordsSize = %s.sizeInBytes();%n", field.camelCaseName());
+                    buffer.printf("_size.addBytes(%s.sizeInBytes());%n", field.camelCaseName());
```

Comment:
Why it is not ```addZeroCopyBytes```?

Reply:
Yeah, my bad. Probably messed this up after renaming.